src/driver/read_fixed.rs

```diff
@@ -0,0 +1,64 @@
+use crate::buf::fixed::FixedBuf;
+use crate::buf::{IoBuf, IoBufMut, Slice};
+use crate::driver::op::{self, Completable};
+use crate::driver::{Op, SharedFd};
+use crate::BufResult;
+
+use std::io;
+
+pub(crate) struct ReadFixed {
```

Comment:
Now I see a little more, I'm confused a little more. 

The fixed read and write operations should work on sockets too, although I don't find liburing tests that expressly prove that. The man pages for fixed read and write say they are like the read and write operations, and those work for socket fds.

This morning, I'm confused by the existing driver in this crate having a 'struct Read<T>' in driver/read.rs and that struct has a buf: T field, but there is also a struct Socket in driver/socket.rs that does not have a buf field.

So at a minimum, I'm wondering if these read_fixed_at and write_fixed_at should be offered in the socket.rs Socket impl too. Or at least a comment saying once this is in for files, we'll want to do the same for sockets at a later time.

Reply:
I've been wondering about how these operatoins apply to sockets too.
I guess for sockets the reads/writes can't have a meaningful offset, so the methods will be named `read_fixed` and `write_fixed`.

Reply:
Yes. Very reasonable. Can be a later PR if you like.